src/components/coin-row/FastTransactionCoinRow.tsx

```diff
@@ -0,0 +1,264 @@
+import {
+  getHumanReadableDate,
+  hasAddableContact,
+} from '@rainbow-me/helpers/transactions';
+import { isValidDomainFormat } from '@rainbow-me/helpers/validators';
+import { useNavigation } from '@rainbow-me/navigation';
+import { RenderProfiler } from '@rainbow-me/performance/utils';
+import Routes from '@rainbow-me/routes';
+import {
+  abbreviations,
+  ethereumUtils,
+  showActionSheetWithOptions,
+} from '@rainbow-me/utils';
+
+const startCase = (string: string) =>
+  string.charAt(0).toUpperCase() + string.slice(1);
+
+const cx = StyleSheet.create({
+  bottomRow: {
+    alignItems: 'center',
+    flex: 1,
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+  },
+  column: {
+    flex: 1,
+    marginLeft: 8,
+  },
+  topRow: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+  },
+  wholeRow: {
+    alignItems: 'center',
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    paddingHorizontal: 19,
+    paddingVertical: 10,
+  },
+});
+
+const BottomRow = ({
+  description,
+  native,
+  status,
+  type,
+  theme,
+}: {
+  description: string;
+  native: any;
+  status: keyof typeof TransactionStatusTypes;
+  type: keyof typeof TransactionTypes;
+  theme: ReturnType<typeof useTheme>;
+}) => {
+  const { colors } = theme;
+  const isFailed = status === TransactionStatusTypes.failed;
+  const isReceived =
+    status === TransactionStatusTypes.received ||
+    status === TransactionStatusTypes.purchased;
+  const isSent = status === TransactionStatusTypes.sent;
+
+  const isOutgoingSwap = status === TransactionStatusTypes.swapped;
+  const isIncomingSwap =
+    status === TransactionStatusTypes.received &&
+    type === TransactionTypes.trade;
+
+  let coinNameColor = colors.dark;
+  if (isOutgoingSwap) coinNameColor = colors.alpha(colors.blueGreyDark, 0.5);
+
+  let balanceTextColor = colors.alpha(colors.blueGreyDark, 0.5);
+  if (isReceived) balanceTextColor = colors.green;
+  if (isSent) balanceTextColor = colors.dark;
+  if (isIncomingSwap) balanceTextColor = colors.swapPurple;
+  if (isOutgoingSwap) balanceTextColor = colors.dark;
+
+  const nativeDisplay = native?.display;
+  const balanceText = nativeDisplay
+    ? [isFailed || isSent ? '-' : null, nativeDisplay].filter(Boolean).join(' ')
+    : '';
+
+  return (
+    <View style={cx.bottomRow}>
+      <Text color={{ custom: coinNameColor || colors.dark }} size="16px">
+        {description}
+      </Text>
+      <Text
+        align="right"
+        color={{ custom: balanceTextColor || colors.dark }}
+        size="16px"
+        weight={isReceived ? 'medium' : undefined}
+      >
+        {balanceText}
+      </Text>
+    </View>
+  );
+};
+
+export default React.memo(function TransactionCoinRow({
+  item,
+  theme,
+}: {
+  item: any;
+  theme: ReturnType<typeof useTheme>;
+}) {
+  const { accountAddress, contact, mainnetAddress } = item;
+  const { navigate } = useNavigation();
```

Comment:
I think this should be also passed. Ideally `onPressTransaction` should be constructed outside